low-level/sensors/impl/generic/node_hw.py

```diff
@@ -636,109 +689,88 @@ def _run_ipmitool_subcommand(self, subcommand, grep_args=None, out_file=subproce
 
         return res, err, retcode
 
-    def _check_channel_error(self, res, err, retcode):
-        # check res present in possible errors or not
-        resource_id = None
-        resource_type = None
-        alert_type = None
-        channel_info = {}
-        channel_status = None
-        epoch_time = str(int(time.time()))
-        raise_fault_resolved_lan = False
-
-        if isinstance(self.active_bmc_if, bytes):
-            self.active_bmc_if = self.active_bmc_if.decode()
-        if isinstance(self.lan_fault,bytes):
-            self.lan_fault = self.lan_fault.decode()
-
-        if self.active_bmc_if == self.ipmi_client.SYSTEM_IF:
-            self._check_kcs_if_alert(err,retcode)
-            channel_info = self.KCS_CHANNEL_INFO
-        elif self.active_bmc_if == self.ipmi_client.LAN_IF or self.lan_fault is None:
-            self._check_lan_if_alert(err,retcode)
-            channel_info = self.LAN_CHANNEL_INFO
-
-        # kcs_interface_alert=True and self.lan_interface_alert=True when fault alert raised
-        # retcode == 0 means ipmitool cmd executed successfully
-        # when both condition satisfied raise falut resolved alert
-        if self.channel_err:
-            if retcode == 0:
-                self.channel_err = False
-                channel_status = "Server BMC is reachable"
-            if self.kcs_channel_err and retcode == 0:
-                alert_type = "fault_resolved"
-                resource_type = "node:bmc:interface:kcs"
-                self.kcs_interface_alert = False
-                self.kcs_channel_err = False
-        if self.lan_cmd_retcode == 0 and self.lan_fault == "fault":
-            alert_type = "fault_resolved"
-            resource_type = "node:bmc:interface:rmcp"
-            self.lan_interface_alert = False
-            raise_fault_resolved_lan = True
-            self.lan_channel_err = False
-            self.lan_fault = None
-            store.put(self.lan_fault,self.LAN_ALERT)
+    def _check_channel_error(self, err, retcode):
+        """Check BMC accessibility for active_interface."""
 
-        if (self.kcs_interface_alert or self.lan_interface_alert) and not self.channel_err:
+        if self.active_bmc_if == system:
+            self.check_kcs_channel(err, retcode)
+        elif self.active_bmc_if in BMCInterface.LAN_IF.value:
+            self.check_lan_channel(err, retcode)
+        if all([self.system_fault, self.lan_fault, self.lanplus_fault]) == "fault":
```

Comment:
how all interface types can get used for a given instance of sspl ?

Reply:
Used one variable for (lan and lanplus) interface as lan_fault and one for system_fault, 
so if initially  active_bmc_if = lan or lanplus and we detect fault, we switched to system interface, and if we detect faut for system interface too then bmc will not be accessible from any channel, so channel_err will be set as True.

Reply:
It will still be either lan or lanplus  interface
Also very hard to relate to this reasoning from code directly, please add comments in code